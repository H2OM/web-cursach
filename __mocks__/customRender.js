import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}