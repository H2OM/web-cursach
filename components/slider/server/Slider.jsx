import Movement from '@/lib/basecomponents/movement/movement';
import '../slider.scss';
import Image from 'next/image';

export default async function Slider() {
    const data = await fetch('http://127.0.0.1/api/slider/get-slider', {method: 'GET', cache: "no-cache"})
    .then(data=>{
        if(!data.ok) {
            return false;
        }
        return data.json();
    }).catch(()=>false);
    const slides = Array.isArray(data) ? data.map((each, i)=>{
        return (
            <div className="Slider__slider__slide" key={each.image+i}>
                <div className="container">
                    <h1 className="title title_slide">{each.title}</h1>
                </div>
                <Image
                    src={"/img/" + each.image}
                    alt="slide"
                    className="Slider__slider__slide__image"
                    width={0}
                    height={0}
                    sizes='100vw'
                    priority={true}
                    quality={100}
                />
            </div>
        )
    }) : <div className='Error'>Ошибка загрузки</div>;
    
    return (
        <section className="Slider">
            <Movement slides={slides}/>
        </section>
    )
}