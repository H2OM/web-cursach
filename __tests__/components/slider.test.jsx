import settingFetchResult from "@/__mocks__/fetchMock";
import { render, screen } from "@testing-library/react";
import Slider from "@/components/slider/server/Slider";
import customRender from "@/__mocks__/customRender";

describe('Slider testing', ()=>{
    it('slider success testing',async ()=>{
        const blocks = ['one1', 'one2', 'one3', 'one4']; 
        const slids = [];
        blocks.forEach((each,i)=>{
            slids.push({
                image: 'some'+i,
                title: each
            });
        });
        
        settingFetchResult(slids);
        const {container} = customRender(await Slider(), {});
        const images = screen.getAllByRole('img');

        expect(images.length).toBe(4);
        for(let block of blocks) {
            expect(await screen.findByText(block)).toBeInTheDocument();
        }

        expect(container).toMatchSnapshot();
    });
    it('slider exeption test', async()=>{
        settingFetchResult(false);
        customRender(await Slider(), {});
        expect(screen.getByText('Ошибка загрузки')).toBeInTheDocument();
    });

});

