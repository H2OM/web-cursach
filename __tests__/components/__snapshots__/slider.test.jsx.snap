// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider testing slider success testing 1`] = `
<div>
  <section
    class="Slider"
  >
    <div
      class="Movement__wrap"
    >
      <div
        class="Movement__slider"
        role="wrap"
        style="transform: translateX(-0%);"
      >
        <div
          class="Slider__slider__slide"
        >
          <div
            class="container"
          >
            <h1
              class="title title_slide"
            >
              one1
            </h1>
          </div>
          <img
            alt="slide"
            class="Slider__slider__slide__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="0"
            sizes="100vw"
            src="/_next/image?url=%2Fimg%2Fsome0&w=3840&q=100"
            srcset="/_next/image?url=%2Fimg%2Fsome0&w=640&q=100 640w, /_next/image?url=%2Fimg%2Fsome0&w=750&q=100 750w, /_next/image?url=%2Fimg%2Fsome0&w=828&q=100 828w, /_next/image?url=%2Fimg%2Fsome0&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2Fsome0&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2Fsome0&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2Fsome0&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2Fsome0&w=3840&q=100 3840w"
            style="color: transparent;"
            width="0"
          />
        </div>
        <div
          class="Slider__slider__slide"
        >
          <div
            class="container"
          >
            <h1
              class="title title_slide"
            >
              one2
            </h1>
          </div>
          <img
            alt="slide"
            class="Slider__slider__slide__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="0"
            sizes="100vw"
            src="/_next/image?url=%2Fimg%2Fsome1&w=3840&q=100"
            srcset="/_next/image?url=%2Fimg%2Fsome1&w=640&q=100 640w, /_next/image?url=%2Fimg%2Fsome1&w=750&q=100 750w, /_next/image?url=%2Fimg%2Fsome1&w=828&q=100 828w, /_next/image?url=%2Fimg%2Fsome1&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2Fsome1&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2Fsome1&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2Fsome1&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2Fsome1&w=3840&q=100 3840w"
            style="color: transparent;"
            width="0"
          />
        </div>
        <div
          class="Slider__slider__slide"
        >
          <div
            class="container"
          >
            <h1
              class="title title_slide"
            >
              one3
            </h1>
          </div>
          <img
            alt="slide"
            class="Slider__slider__slide__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="0"
            sizes="100vw"
            src="/_next/image?url=%2Fimg%2Fsome2&w=3840&q=100"
            srcset="/_next/image?url=%2Fimg%2Fsome2&w=640&q=100 640w, /_next/image?url=%2Fimg%2Fsome2&w=750&q=100 750w, /_next/image?url=%2Fimg%2Fsome2&w=828&q=100 828w, /_next/image?url=%2Fimg%2Fsome2&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2Fsome2&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2Fsome2&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2Fsome2&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2Fsome2&w=3840&q=100 3840w"
            style="color: transparent;"
            width="0"
          />
        </div>
        <div
          class="Slider__slider__slide"
        >
          <div
            class="container"
          >
            <h1
              class="title title_slide"
            >
              one4
            </h1>
          </div>
          <img
            alt="slide"
            class="Slider__slider__slide__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="0"
            sizes="100vw"
            src="/_next/image?url=%2Fimg%2Fsome3&w=3840&q=100"
            srcset="/_next/image?url=%2Fimg%2Fsome3&w=640&q=100 640w, /_next/image?url=%2Fimg%2Fsome3&w=750&q=100 750w, /_next/image?url=%2Fimg%2Fsome3&w=828&q=100 828w, /_next/image?url=%2Fimg%2Fsome3&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2Fsome3&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2Fsome3&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2Fsome3&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2Fsome3&w=3840&q=100 3840w"
            style="color: transparent;"
            width="0"
          />
        </div>
      </div>
      <nav
        class="Movement__nav"
      >
        <div
          class="Movement__nav__point Movement__nav__point_select"
          role="navRadio"
        />
        <div
          class="Movement__nav__point "
          role="navRadio"
        />
        <div
          class="Movement__nav__point "
          role="navRadio"
        />
        <div
          class="Movement__nav__point "
          role="navRadio"
        />
      </nav>
    </div>
  </section>
</div>
`;
